src/iexamine.cpp

```diff
@@ -622,55 +622,33 @@ void iexamine::cardreader(player *p, map *m, const tripoint &examp)
-        if (using_electrohack || using_fingerhack) {
-            p->moves -= 500;
-            p->practice( skill_computer, 20 );
-            int success = rng(p->skillLevel( skill_computer ) / 4 - 2, p->skillLevel( skill_computer ) * 2);
-            success += rng(-3, 3);
-            if (using_fingerhack) {
-                success++;
-            }
-            if (p->int_cur < 8) {
-                success -= rng(0, int((8 - p->int_cur) / 2));
-            } else if (p->int_cur > 8) {
-                success += rng(0, int((p->int_cur - 8) / 2));
-            }
-            if (success < 0) {
-                add_msg(_("You cause a short circuit!"));
-                if (success <= -5) {
-                    if (using_electrohack) {
-                        add_msg(m_bad, _("Your electrohack is ruined!"));
-                        p->use_amount("electrohack", 1);
-                    } else {
-                        add_msg(m_bad, _("Your power is drained!"));
-                        p->charge_power(-rng(0, p->power_level));
-                    }
-                }
+        switch (hack_attempt(p)) {
+            case HACK_FAIL:
                 m->ter_set(examp, t_card_reader_broken);
-            } else if (success < 6) {
+                break;
+            case HACK_NOTHING:
                 add_msg(_("Nothing happens."));
-            } else {
-                add_msg(_("You activate the panel!"));
-                add_msg(m_good, _("The nearby doors slide into the floor."));
-                m->ter_set(examp, t_card_reader_broken);
-                tripoint tmp = examp;
-                int &i = tmp.x;
-                int &j = tmp.y;
-                for (i = examp.x - 3; i <= examp.x + 3; i++) {
-                    for (j = examp.y - 3; j <= examp.y + 3; j++) {
-                        if (m->ter(tmp) == t_door_metal_locked) {
-                            m->ter_set(tmp, t_floor);
+                break;
+            case HACK_SUCCESS:
+                {
+                    add_msg(_("You activate the panel!"));
+                    add_msg(m_good, _("The nearby doors slide into the floor."));
+                    m->ter_set(examp, t_card_reader_broken);
+                    tripoint tmp = examp;
+                    int &i = tmp.x;
+                    int &j = tmp.y;
+                    for (i = examp.x - 3; i <= examp.x + 3; i++) {
+                        for (j = examp.y - 3; j <= examp.y + 3; j++) {
+                            if (m->ter(tmp) == t_door_metal_locked) {
+                                m->ter_set(tmp, t_floor);
+                            }
                         }
                     }
                 }
-            }
-        } else {
-            add_msg(m_info, _("Looks like you need a %s."), item::nname( card_type ).c_str());
+                break;
+            case HACK_UNABLE:
+                add_msg(m_info, _("Looks like you need a %s."), item::nname( card_type ).c_str());
```

Comment:
The other two functions outright tell you the missing item, which makes this inconsistent. Also, just picking up an electrohack (without even knowing what it is) would suddenly give the information, that's even stranger. I like the idea of the additional hint, but maybe include a hint that it needs an item, otherwise player might be searching for the "hack this" key.